HW_11/task2.py

```diff
@@ -0,0 +1,62 @@
+# 1. Создайте класс Матрица. Добавьте методы для:
+#    - вывода на печать,
+#    - проверку на равенство,
+#    - сложения,
+#    - *умножения матриц.
+
+class Matrix:
+
+    def __init__(self, matrix: list[list[int, float]]):
+        self.matrix = matrix
+        self.rows = len(matrix)
+        self.columns = len(matrix[0])
+
+    def __add__(self, other):
+        if isinstance(other, Matrix):
+            if self.columns == other.columns and self.rows == other.rows:
+                result = [[] for _ in range(self.rows)]
+                for i in range(self.rows):
+                    for j in range(self.columns):
+                        result[i].append(self.matrix[i][j] + other.matrix[i][j])
+                return Matrix(result)
+
+    def __mul__(self, other):
+        if isinstance(other, Matrix):
+            if self.columns == other.rows:
+                result = [[sum(a * b for a, b in zip(self_row, other_col))
+                           for other_col in zip(*other.matrix)]
+                          for self_row in self.matrix]
+            elif self.rows == other.columns:
+                result = [[sum(a * b for a, b in zip(self_col, other_row))
+                           for self_col in zip(*self.matrix)]
+                          for other_row in other.matrix]
```

Comment:
Избегайте повторений.
```suggestion
            if self.columns == other.rows and self.rows == other.columns:
                result = [[sum(a * b for a, b in zip(self_row, other_col))
                           for other_col in zip(*other.matrix)]
                          for self_row in self.matrix]
```